### PR TITLE
implement response discriminator checker

### PR DESCRIPTION
**Pull request by ciprian-tibulca**

### CODE REVIEW

**Review by blva — APPROVED** [review]

File: checker/check-request-discriminator-updated.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+					Id:          messageId,
+					Level:       INFO,
+					Text:        fmt.Sprintf(config.i18n(messageId), a...),
+					Operation:   operation,
+					OperationId: operationItem.Revision.OperationID,
+					Path:        path,
+					Source:      source,
+				})
+			}
+
```
Comment (reuvenharrison) [inline checker/check-request-discriminator-updated.go:39]: Like :-)

File: checker/check-request-discriminator-updated.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
+package checker
+
+import (
+	"fmt"
+
+	"github.com/tufin/oasdiff/diff"
+)
+
+func RequestDiscriminatorUpdatedCheck(diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap, config Config) Changes {
```
Comment (reuvenharrison) [inline checker/check-request-discriminator-updated.go:9]: Add checker to defaultChecks

Comment (ciprian-tibulca) [reply checker/check-request-discriminator-updated.go:9]: good catch, thank you.
Updated defaultChecks

File: checker/localizations_src/en/messages.yaml [hunk 3]
```diff
@@ -220,11 +220,35 @@ response-body-all-of-added: added %s to the response body 'allOf' list for the r
 response-body-all-of-removed: removed %s from the response body 'allOf' list for the response status %s
 response-property-all-of-added: added %s to the %s response property 'allOf' list for the response status %s
 response-property-all-of-removed: removed %s from the %s response property 'allOf' list for the response status %s
+response-body-discriminator-added: added response discriminator for the response status %s
+response-property-discriminator-added: added discriminator to %s response property for the response status %s
+response-body-discriminator-removed: removed response discriminator for the response status %s
+response-property-discriminator-removed: removed discriminator to %s response property for the response status %s
```
Comment (reuvenharrison) [inline checker/localizations_src/en/messages.yaml]: to -> from

File: checker/localizations_src/en/messages.yaml [hunk 4]
```diff
@@ -220,11 +220,35 @@ response-body-all-of-added: added %s to the response body 'allOf' list for the r
...
+response-property-discriminator-mapping-deleted: removed %s discriminator mapping keys from the %s response property for the response status %s
+response-body-discriminator-mapping-changed: mapped value for key %s changed from %s to %s from the response discriminator for the response status %s
+response-property-discriminator-mapping-changed: mapped value for discriminator key %s changed from %s to %s for %s response property for the response status %s
+request-body-discriminator-added: added request schema discriminator
+request-property-discriminator-added: added discriminator to %s request property
+request-body-discriminator-removed: removed request schema discriminator
+request-property-discriminator-removed: removed discriminator to %s request property
+request-body-discriminator-property-name-changed: request schema discriminator property name changed from %s to %s
+request-property-discriminator-property-name-changed: request schema discriminator property name changed for %s request property from %s to %s
+request-body-discriminator-mapping-added: added %s mapping keys to the request schema discriminator
```
Comment (reuvenharrison) [inline checker/localizations_src/en/messages.yaml]: for consistency, I recommend to standardize on either "schema discriminator" or "discriminator"

File: checker/localizations_src/en/messages.yaml (hunk 3 above)
Comment (ciprian-tibulca) [reply checker/localizations_src/en/messages.yaml]: done, thank you

File: checker/localizations_src/en/messages.yaml (hunk 4 above)
Comment (ciprian-tibulca) [reply checker/localizations_src/en/messages.yaml]: sure, using "discriminator" in all messages now

File: checker/check-request-discriminator-updated_test.go [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+	s2, err := open("../data/checker/request_property_discriminator_mapping_changed.yaml")
+	require.NoError(t, err)
+
+	d, osm, err := diff.GetWithOperationsSourcesMap(getConfig(), s1, s2)
+	require.NoError(t, err)
+	errs := checker.CheckBackwardCompatibilityUntilLevel(singleCheckConfig(checker.RequestDiscriminatorUpdatedCheck), d, osm, checker.INFO)
+
+	require.Len(t, errs, 5)
+
+	require.ElementsMatch(t, []checker.ApiChange{
```
Comment (reuvenharrison) [inline checker/check-request-discriminator-updated_test.go:132]: Like :-)

**Pull request merged**